File: term4/Kurs-projektowania-aplikacji-z-bazami-danych/project/Store/Store.Infrastructure/Repositories/ShipmentRepository/TestShipmentRepository.cs
using Store.Common;
using Store.Domain.Model.Shared;
using Store.Domain.Model.Shipment;
using System.Collections.Generic;

namespace Store.Infrastructure.Repositories.ShipmentRepository
{
    public class TestShipmentRepository : GenericTestRepository<Shipment>, IShipmentRepository
    {
        public TestShipmentRepository() : this(new List<Shipment> {
            new Shipment { Id = 1,
                DeliveryAddress = new Address { Country = "test", City = "test", Detail = "test", ZipCode = "12-345"},
                Status = ShipmentStatus.Ready,
                Type = ShipmentType.Air },
            new Shipment { Id = 2,
                DeliveryAddress = new Address { Country = "test", City = "test", Detail = "test", ZipCode = "12-345"},
                Status = ShipmentStatus.Delivered,
                Type = ShipmentType.Land },
            new Shipment { Id = 3,
                DeliveryAddress = new Address { Country = "test", City = "test", Detail = "test", ZipCode = "12-345"},
                Status = ShipmentStatus.InProgress,
                Type = ShipmentType.Naval },
        }) { }

        private TestShipmentRepository(ICollection<Shipment> shipments) : base(shipments) { }
    }
}

File: term4/database-applications/project-old/src/main/kotlin/dbapp/project/infrastructure/test/TestShipmentRepository.kt
package dbapp.project.infrastructure.test

import dbapp.project.common.GenericTestRepository
import dbapp.project.domain.shared.Address
import dbapp.project.domain.shipment.Shipment
import dbapp.project.domain.shipment.ShipmentRepository
import dbapp.project.domain.shipment.ShipmentStatus
import dbapp.project.domain.shipment.ShipmentType

class TestShipmentRepository : GenericTestRepository<Shipment>(
        mutableSetOf(
                Shipment(1,
                        ShipmentType("naval"),
                        Address("Test", "Test", "Test 21/37", "12-345"),
                        ShipmentStatus("in progress")),
                Shipment(2,
                        ShipmentType("air"),
                        Address("Test", "Test", "Test 21/37", "12-345"),
                        ShipmentStatus("in progress")),
                Shipment(3,
                        ShipmentType("air"),
                        Address("Test", "Test", "Test 21/37", "12-345"),
                        ShipmentStatus("delivered"))
        )
), ShipmentRepository
